File: ProfileBook.API/Models/Report.cs
namespace ProfileBook.API.Models
{
    public class Report
    {
        public int ReportId { get; set; } // Primary Key 
        public int ReportedUserId { get; set; }
        public int ReportingUserId { get; set; }
        public string? Reason { get; set; }
        public DateTime TimeStamp { get; set; } = DateTime.Now; 
    }
}

File: after_signalR/ProfileBook.UI/src/app/services/report.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environment';
import { Report } from '../models';

@Injectable({
  providedIn: 'root',
})
export class ReportService {
  private baseUrl = `${environment.apiUrl}/Report`;

  constructor(private http: HttpClient) {}

  getReports(): Observable<Report[]> {
    return this.http.get<Report[]>(this.baseUrl);
  }

  reportUser(reportedUserId: number, reportingUserId: number, reason?: string): Observable<unknown> {
    return this.http.post(this.baseUrl, {
      reportedUserId,
      reportingUserId,
      reason: reason ?? undefined,
    });
  }
}
